### PR TITLE
Improve --clean option

### PR DESCRIPTION
**Pull request by soumeh01**
## Fixes
- https://github.com/Open-CMSIS-Pack/devtools/issues/1780

## Changes
- The change includes removing dependency on convert command

## Checklist

- [x] 🤖 This change is covered by unit tests as required.
- [x] 🤹 All required manual testing has been performed.
- [x] 🛡️ Security impacts have been considered.
- [x] 📖 All documentation updates are complete.
- [x] 🧠 This change does not change third-party dependencies

### CODE REVIEW

File: pkg/builder/csolution/builder.go [hunk 1]
```diff
@@ -735,3 +690,111 @@ func (b CSolutionBuilder) hasRebuildNode(idxFilePath string) (bool, error) {
...
+
+	// lambda function to remove dir using cmake
+	removeDirectory := func(dir string) error {
+		if _, err := os.Stat(dir); os.IsNotExist(err) {
+			return nil
+		}
+		args := []string{"-E", "remove_directory", dir}
+		_, err = b.Runner.ExecuteCommand(vars.CmakeBin, false, args...)
+		return err
+	}
```
Comment (brondani) [inline pkg/builder/csolution/builder.go:728]: Can we avoid using CMake for cleaning? This would potentially fail in KSC IDE.

File: README.md
Comment (jkrech) [inline README.md:1]: Should we still maintain this here?
I thinks cbuild invocation is documented here: 
https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/build-tools.md#cbuild-invocation

**Review by jkrech — APPROVED** [review]
LGTM.

File: pkg/builder/csolution/builder.go [hunk 2]
```diff
@@ -735,3 +690,95 @@ func (b CSolutionBuilder) hasRebuildNode(idxFilePath string) (bool, error) {
...
+	outputDir := b.Options.Output
+
+	var tmpDir string
+	if !b.Options.UseCbuild2CMake {
+		// Use default path when --cbuildgen option is used
+		tmpDir = filepath.Join(filepath.Dir(b.InputFile), outputDir, "tmp")
+	} else {
+		tmpDir, err = utils.GetTmpDir(b.InputFile, outputDir)
+		if err != nil {
+			return err
```
Comment (brondani) [inline pkg/builder/csolution/builder.go:710]: I believe the non existence of the `tmp` directory should not be a reason to return error.

File: pkg/builder/csolution/builder.go [hunk 3]
```diff
@@ -735,3 +690,95 @@ func (b CSolutionBuilder) hasRebuildNode(idxFilePath string) (bool, error) {
...
+			seplen = len(cleanMsg)
+			utils.PrintSeparator("-", seplen)
+		}
+		utils.LogStdMsg(cleanMsg)
+
+		idxFile, err := b.getIdxFilePath()
+		if err == nil {
+			outDir, err := utils.GetOutDir(idxFile, context)
+			if err != nil {
+				log.Error("error cleaning '" + context + "'")
```
Comment (brondani) [inline pkg/builder/csolution/builder.go:734]: Similarly to `tmp`, the absence of `out` should not return error.

**Review by brondani — APPROVED** [review]
LGTM

**Pull request merged**